lib/rules/number-leading-zero/index.js

```diff
@@ -56,16 +56,30 @@ const rule = function(expectation) {
 
         // Check leading zero
         if (expectation === "always") {
-          const match = /(?:\D|^)(\.\d+)/.exec(valueNode.value);
+          const regexp = /((?:\D|^))(\.\d+)/;
+          const match = regexp.exec(valueNode.value);
 
           if (match === null) {
             return;
           }
 
+          if (context.fix) {
+            if (node.value) {
+              node.value = node.value.replace(regexp, "$10$2");
+              return;
+            }
+
+            if (node.params) {
```

Comment:
From coverage reports, it's never true:
https://coveralls.io/builds/13380484/source?filename=lib%2Frules%2Fnumber-leading-zero%2Findex.js#L72

Reply:
I haven't used it in the beginning, but then it was not working for the example `code: "@media (min-width: .01em)"`, as it does not have a node value, just node params. So maybe I am missing something here. It's my first PR to stylelint. Can you help me on this one?

Reply:
@jeddy3 can you have a look at this one too? This one I would actually need quite quick :)

Reply:
@gucong3000 Can you expand on your comment? 

I believe [the rule also checks](https://github.com/pkuczynski/stylelint/blob/bf5ee41da9b4db384392f9874e06b252bedb2e67/lib/rules/number-leading-zero/index.js#L27-L33) _media features_, as well as _declarations_. `atRule` nodes have a `params` property, rather than a `value` one. 

Are you suggesting the use of [postcss-media-query-parser](https://github.com/dryoma/postcss-media-query-parser) to parse out the _value_ of the _media feature_, thus normalising the behaviour to only changing _values_ regardless of whether it's a _declaration_ or _media feature_? I guess this would be preferable as it would scope the regex to the smallest possible area.

Reply:
> This one I would actually need quite quick :)

I believe the core team is spread across different time zones and so the review process can take a few days. I'll hold off doing a release until the weekend though, in the hopes that your two PRs make it in.

Reply:
Yes, this rule checks both properties and media queries. I am not Post CSS expert, so I would need your advice how to improve this code...

I understand the time zones issue. It would be awesome if this change could hit the nearest release...

Reply:
What I'm worried about is that using the regular replacement string directly, may not be able to cope with unknown complexities.

Reply:
And here's the inner of `valueParser(value).walk()`.

Reply:
Not sure if regexp could break anything here, but you are right, you never know in regexp :) I do not understand the comment about `valueParser(value).walk()`. You would have to be more specific to help me go through this...

Reply:
> I do not understand the comment about `valueParser(value).walk()`.
https://github.com/TrySound/postcss-value-parser#valueparserwalknodes-callback-bubble
callback function will recursive calls multiple times.
This line of code is not run once for each `node`.

Reply:
@gucong3000 Can you push your proposed changes to @pkuczynski branch, please? I think that'll help us get this PR in.

@pkuczynski I believe @gucong3000 is pointing out that your fix happens [within the postcss-value-parser `walk` loop](https://github.com/pkuczynski/stylelint/blob/bf5ee41da9b4db384392f9874e06b252bedb2e67/lib/rules/number-leading-zero/index.js#L43-L125), and you are therefore setting `node.value` for _every `word` node_ rather than just once.